Content.Server/Atmos/EntitySystems/GasMinerSystem.cs

```diff
@@ -21,70 +21,74 @@ public override void Initialize()
         SubscribeLocalEvent<GasMinerComponent, AtmosDeviceUpdateEvent>(OnMinerUpdated);
     }
 
-    private void OnMinerUpdated(Entity<GasMinerComponent> ent, ref AtmosDeviceUpdateEvent args)
+    private void ProcessMinerState(Entity<GasMinerComponent> ent, GasMinerState newState, float? possibleNewStoredAmount = null)
     {
-        var miner = ent.Comp;
-        var oldState = miner.MinerState;
-        float toSpawn;
+        var minerComponent = ent.Comp;
 
-        if (!GetValidEnvironment(ent, out var environment) || !Transform(ent).Anchored)
-        {
-            miner.MinerState = GasMinerState.Disabled;
-        }
-        // SpawnAmount is declared in mol/s so to get the amount of gas we hope to mine, we have to multiply this by
-        // how long we have been waiting to spawn it and further cap the number according to the miner's state.
-        else if ((toSpawn = CapSpawnAmount(ent, miner.SpawnAmount * args.dt, environment)) == 0)
-        {
-            miner.MinerState = GasMinerState.Idle;
-        }
-        else
-        {
-            miner.MinerState = GasMinerState.Working;
+        if (minerComponent.MinerState != newState)
+        { minerComponent.MinerState = newState; DirtyField(ent.Owner, minerComponent, nameof(minerComponent.MinerState)); }
 
-            // Time to mine some gas.
-            var merger = new GasMixture(1) { Temperature = miner.SpawnTemperature };
-            merger.SetMoles(miner.SpawnGas, toSpawn);
-            _atmosphereSystem.Merge(environment, merger);
-        }
-
-        if (miner.MinerState != oldState)
+        if (possibleNewStoredAmount is { } newStoredAmount && Math.Abs(minerComponent.LastReplicatedStoredAmount - newStoredAmount) >= float.Epsilon)
         {
-            Dirty(ent);
+            minerComponent.StoredAmount = newStoredAmount;
+            minerComponent.LastReplicatedStoredAmount = newStoredAmount;
+            DirtyField(ent.Owner, minerComponent, nameof(minerComponent.StoredAmount));
         }
     }
 
-    private bool GetValidEnvironment(Entity<GasMinerComponent> ent, [NotNullWhen(true)] out GasMixture? environment)
+    private void OnMinerUpdated(Entity<GasMinerComponent> ent, ref AtmosDeviceUpdateEvent args)
     {
-        var (uid, miner) = ent;
-        var transform = Transform(uid);
-        var position = _transformSystem.GetGridOrMapTilePosition(uid, transform);
+        var minerComponent = ent.Comp;
 
-        // Treat space as an invalid environment
-        if (_atmosphereSystem.IsTileSpace(transform.GridUid, transform.MapUid, position))
-        {
-            environment = null;
-            return false;
-        }
+        if (!GetValidEnvironment(ent, out var environment, out var transform, out var minerTilePosition) || !transform.Anchored)
+        { ProcessMinerState(ent, GasMinerState.Disabled); return; }
```

Comment:
Please use line breaks, similar case throughout the PR
```suggestion
        if (!GetValidEnvironment(ent, out var environment, out var transform, out var minerTilePosition) ||
            !transform.Anchored)
        {
            ProcessMinerState(ent, GasMinerState.Disabled);
            return;
        }
```